nautobot/core/management/commands/pre_migrate.py

```diff
@@ -75,6 +86,115 @@ def check_configcontext_uniqueness():
         )
 
 
+def check_permissions_constraints(command):
+    """
+    Check for permission constraints that are referencing an object field
+    that will be migrated to a different model/field/value in 2.0 migrations.
+
+    Objects that will be deleted in 2.0 are:
+    - dcim.DeviceRole
+    - dcim.RackRole
+    - ipam.Aggregate
+    - ipam.Role
+
+    Objects that will possibly have fields migrated to a different value in 2.0 are:
+    - ipam.IPAddress
+    - ipam.Prefix
+    - ipam.Service
+    - extras.CustomField
+    - extras.GitRepository
+    - extras.Job
+    - extras.ObjectPermission
+    - extras.ScheduledJob
+    - extras.TaggedItem
+    """
+
+    deleted_models = [
+        Aggregate,
+        DeviceRole,
+        RackRole,
+        Role,
+    ]
+
+    pk_change_models = [
+        IPAddress,
+        Prefix,
+        ObjectPermission,
+        TaggedItem,
+    ]
+
+    field_change_models = {
```

Comment:
Should this or something similar cover all of the field changes documented in 2.0 in `nautobot/docs/user-guide/administration/upgrading/from-v1/tables/v2-database-renamed-fields.yaml` and/or `v2-database-behavior-changes.yaml` and/or `v2-database-removed-fields.yaml`?

E.g. `Vlan.group` --> `VLAN.vlan_group`, `ConfigContext.regions` -> `ConfigContext.locations`, etc etc.

Reply:
That's what I was hoping to accomplish. Do the best we can here but also document that it's not 100%. The field renames and region/site shouldn't be too problematic because new random UUIDs aren't being assigned but for Aggregate, Prefix and IPAddress it's going to be important to document the old primary keys.

Reply:
This caught me, I read through the migration guide and missed the group -> vlan_group or the temrination_a|z -> circuit_termination_a|z. Also an issue with `DeviceType.instances` -> `DeviceType.devices`.